src/pk-main.c

```diff
@@ -182,7 +182,16 @@ main (int argc, char *argv[])
 	syslog (LOG_DAEMON | LOG_DEBUG, "daemon start");
 
 	/* after how long do we timeout? */
-	exit_idle_time = g_key_file_get_integer (conf, "Daemon", "ShutdownTimeout", NULL);
+	exit_idle_time = g_key_file_get_integer (conf, "Daemon", "ShutdownTimeout", &error);
+	/* THIS COMMENT IS A TSUNAMI STONE
+         * The automatic shutdown timeout is necessary to avoid very high
+         * memory usage with the dnf backend.  If you want to remove this
+         * timeout, please study the git history and be sure that you are not
+         * regressing Redhat bugzilla #1354074 (again). */
+	if (error != NULL) {
+		exit_idle_time = 300;
+		g_clear_error(&error);
```

Comment:
Yes, this is the right thing to do. You have to manually free the error and set it to NULL here, despite the g_autoptr, because the error is reused later on in this function and it's a bug to fail to unset the error before passing it to function that might set it. g_error_free() is not enough, so g_clear_error() is best.

However, code style fix:

```suggestion
		g_clear_error (&error);
```